juju/juju
# Add network interfaces to show-machine

Replace ip-addresses list with network-interfaces hash in status and show-machine output

----

## Description of change

This change adds more networking-related output to show-machine.

The network-interfaces hash include the following elements per machine:
 * addresses
 * gateway
 * nameservers
 * space

## Note on tests

Existing tests have been updated and pass. Tests where improvement is required in a next iteration are noted as TODO items. Under the circumstances of getting this out before 2017-02-10, I'm submitting the PR as-is, with the intention that tests should be expanded.

## QA steps

 * `juju status --format yaml` and `juju show-machine <id>`
 * Add more IP addresses to a NIC and see them appear in output
 * Use a space and see it appear in output

## Documentation changes

`juju status --format [yaml|json]` and `juju show-machine` now have a network-interfaces hash, which includes the existing ip-addresses list.

## Bug reference

lp:1653997
